src/app/icd/client/DefaultICDClientInfoPersistentStorage.cpp

```diff
@@ -0,0 +1,347 @@
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultICDClientStorage::Delete(ICDStorage & aStorage, size_t aIndex, Crypto::Aes128KeyHandle & aSharedKey)
+{
+    VerifyOrReturnError(aStorage.mpClientInfoStore != nullptr, CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(aStorage.mpClientInfoStore->SyncDeleteKeyValue(mpICDStorageKeyDelegate->GetKey(aIndex).KeyName()));
+    return DeleteKey(aStorage.mpKeyStore, aSharedKey);
+}
+
+CHIP_ERROR DefaultICDClientStorage::DeleteKey(Crypto::SymmetricKeystore * apKeyStore, Crypto::Aes128KeyHandle & aSharedKey)
+{
+    VerifyOrReturnError(apKeyStore != nullptr, CHIP_ERROR_INTERNAL);
+    apKeyStore->DestroyKey(aSharedKey);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultICDClientStorage::DeleteEntry(ScopedNodeId aPeerNode)
+{
+    ICDStorage * storage = FindStorage(aPeerNode.GetFabricIndex());
+    VerifyOrReturnError(storage != nullptr, CHIP_ERROR_NOT_FOUND);
+    size_t maxCount = mpICDStorageKeyDelegate->MaxKeyCounter();
+    for (size_t index = 0; index < maxCount; index++)
+    {
+        ICDClientInfo currentICDClientInfo;
+        CHIP_ERROR err = Load(*storage, index, currentICDClientInfo);
+        if (err == CHIP_NO_ERROR)
+        {
+            if (aPeerNode.GetNodeId() == currentICDClientInfo.mPeerNode.GetNodeId())
+            {
+                Delete(*storage, index, currentICDClientInfo.mSharedKey);
+                break;
+            }
+        }
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultICDClientStorage::DeleteAllEntries(FabricIndex aFabricIndex)
+{
+    ICDStorage * storage = FindStorage(aFabricIndex);
+    VerifyOrReturnError(storage != nullptr, CHIP_ERROR_NOT_FOUND);
+    size_t maxCount = mpICDStorageKeyDelegate->MaxKeyCounter();
+    for (size_t index = 0; index < maxCount; index++)
+    {
+        ICDClientInfo currentICDClientInfo;
+        CHIP_ERROR err = Load(*storage, index, currentICDClientInfo);
+        if (err == CHIP_NO_ERROR)
+        {
+            Delete(*storage, index, currentICDClientInfo.mSharedKey);
+        }
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultICDClientStorage::AddStorage(ICDStorage && aStorage)
+{
+    VerifyOrReturnError(aStorage.IsValid(), CHIP_ERROR_INVALID_ARGUMENT);
+    mStorages.push_back(std::move(aStorage));
+    return CHIP_NO_ERROR;
+}
+
+void DefaultICDClientStorage::RemoveStorage(FabricIndex aFabricIndex)
+{
+    DeleteAllEntries(aFabricIndex);
+    for (auto storageIterator = mStorages.begin(); storageIterator != mStorages.end(); storageIterator++)
+    {
+        if (storageIterator->mFabricIndex == aFabricIndex)
+        {
+            mStorages.erase(storageIterator);
+            break;
+        }
+    }
+}
+
+bool DefaultICDClientStorage::ValidateCheckInPayload(const ByteSpan & aPayload, ICDClientInfo & aClientInfo)
+{
+    // TODO: Need to implement default decription code using CheckinMessage::ParseCheckinMessagePayload
+    return false;
+}
+
+size_t DefaultICDClientStorage::Size()
```

Comment:
That said, why are DefaultICDClientStorage methods in _here_, not in DefaultICDClientStorage.cpp?